### PR TITLE
Upload files

### PR DESCRIPTION
**Pull request by maksimfedin**

### CODE REVIEW

File: service-node/backend/src/main/scala/org/enterprisedlt/fabric/service/node/rest/UploadHandler.scala [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+/**
+ * @author Alexey Polubelov
+ */
+
+
+class UploadHandler(
+    multipartConfig: MultipartConfigElement,
+    endpoints: Array[FileUploadEndpoint],
+    createCodec: () => ServerCodec,
```
Comment (apolubelov) [inline service-node/backend/src/main/scala/org/enterprisedlt/fabric/service/node/rest/UploadHandler.scala]: seems to be unused

File: service-node/backend/src/main/scala/org/enterprisedlt/fabric/service/node/rest/UploadHandler.scala [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            baseRequest.setHandled(true)
+        }
+
+    }
+
+    private def processRequest(
+        endpoints: Array[FileUploadEndpoint],
+        request: HttpServletRequest, response: HttpServletResponse
+    ): Unit = {
+        if (endpoints.exists(_.uri == request.getPathInfo)) {
```
Comment (apolubelov) [inline service-node/backend/src/main/scala/org/enterprisedlt/fabric/service/node/rest/UploadHandler.scala]: you already did this check, consider reuse

File: service-node/backend/src/main/scala/org/enterprisedlt/fabric/service/node/rest/UploadHandler.scala [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+                    val outputFile = outputDir.resolve(filename)
+                    val is = part.getInputStream
+                    val os = Files.newOutputStream(outputFile, StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING)
+                    IO.copy(is, os)
+                    logger.info(s"Saved Part ${part.getName} to ${outputFile.toString}")
+                    is.close()
+                    os.close()
+                }
+            }
+        }.toEither match {
```
Comment (apolubelov) [inline service-node/backend/src/main/scala/org/enterprisedlt/fabric/service/node/rest/UploadHandler.scala:60]: consider merge this match with one in calling method

File: service-node/backend/src/main/scala/org/enterprisedlt/fabric/service/node/Util.scala [hunk 4]
```diff
@@ -213,6 +213,15 @@ object Util {
     //=========================================================================
     def mkDirs(path: String): Boolean = new File(path).mkdirs()
 
+
+    def ensureDirExists(path: String): Unit = {
+        val dir = new File(path)
+        if (!dir.exists()) {
```
Comment (apolubelov) [inline service-node/backend/src/main/scala/org/enterprisedlt/fabric/service/node/Util.scala]: unnecessary check, call to this method is equal to call to `Util.mkDirs`

File: service-node/backend/src/main/scala/org/enterprisedlt/fabric/service/node/ServiceNode.scala [hunk 5]
```diff
@@ -135,6 +139,21 @@ object ServiceNode extends App {
             )
         )
 
+        val fileEnpointContext = new ContextHandler("/")
+        val tmpLocation: String = "/tmp/files/" // the directory location where files will be stored
+        val maxFileSize: Int = 100 * 1024 * 1024 // the maximum size allowed for uploaded files
+        val maxRequestSize: Int = 100 * 1024 * 1024 // the maximum size allowed for multipart/form-data requests
+        val fileSizeThreshold: Int = 1024 // the size threshold after which files will be written to disk
+        val multipartConfig: MultipartConfigElement = new MultipartConfigElement(tmpLocation, maxFileSize, maxRequestSize, fileSizeThreshold);
```
Comment (apolubelov) [inline service-node/backend/src/main/scala/org/enterprisedlt/fabric/service/node/ServiceNode.scala]: put the vars above to `UploadHandler` contructor args, initialize `multipartConfig` in `UploadHandler`

File: service-node/frontend/src/main/scala/org/enterprisedlt/fabric/service/node/page/Dashboard.scala [hunk 6]
```diff
@@ -283,7 +293,45 @@ object Dashboard {
                                                 )
                                             )
                                         )
-                                )
+                                ),
+                                PageAction(
+                                    name = "Upload",
+                                    id = "upload-contract",
+                                    actionForm = progress =>
+                                        <.form(
```
Comment (apolubelov) [inline service-node/frontend/src/main/scala/org/enterprisedlt/fabric/service/node/page/Dashboard.scala]: replace with `div` or `Seq(...).toTagMod`

**Review by apolubelov — CHANGES_REQUESTED** [review]

File: service-node/backend/src/main/scala/org/enterprisedlt/fabric/service/node/rest/UploadHandler.scala (hunk 1 above)
Comment (maksimfedin) [reply service-node/backend/src/main/scala/org/enterprisedlt/fabric/service/node/rest/UploadHandler.scala]: deleted.

File: service-node/backend/src/main/scala/org/enterprisedlt/fabric/service/node/rest/UploadHandler.scala (hunk 2 above)
Comment (maksimfedin) [reply service-node/backend/src/main/scala/org/enterprisedlt/fabric/service/node/rest/UploadHandler.scala]: done.

File: service-node/backend/src/main/scala/org/enterprisedlt/fabric/service/node/Util.scala (hunk 4 above)
Comment (maksimfedin) [reply service-node/backend/src/main/scala/org/enterprisedlt/fabric/service/node/Util.scala]: ensureDirExists removed.

File: service-node/backend/src/main/scala/org/enterprisedlt/fabric/service/node/ServiceNode.scala (hunk 5 above)
Comment (maksimfedin) [reply service-node/backend/src/main/scala/org/enterprisedlt/fabric/service/node/ServiceNode.scala]: done.

File: service-node/frontend/src/main/scala/org/enterprisedlt/fabric/service/node/page/Dashboard.scala (hunk 6 above)
Comment (maksimfedin) [reply service-node/frontend/src/main/scala/org/enterprisedlt/fabric/service/node/page/Dashboard.scala]: done.

File: service-node/backend/src/main/scala/org/enterprisedlt/fabric/service/node/rest/UploadHandler.scala (hunk 3 above)
Comment (maksimfedin) [reply service-node/backend/src/main/scala/org/enterprisedlt/fabric/service/node/rest/UploadHandler.scala:60]: done.

**Review by apolubelov — APPROVED** [review]

**Pull request merged**